pkg/devfile/generator/generators.go

```diff
@@ -199,29 +280,74 @@ func GetDeployment(devfileObj parser.DevfileObj, deployParams DeploymentParams)
 	if err != nil {
 		return nil, err
 	}
-	podTemplateSpec, err := getPodTemplateSpec(globalAttributes, components, podTemplateSpecParams)
+
+	podTemplateSpec, err := getPodTemplateSpec(podTemplateSpecParams)
 	if err != nil {
 		return nil, err
 	}
-	deploySpecParams := deploymentSpecParams{
-		PodTemplateSpec:   *podTemplateSpec,
-		PodSelectorLabels: deployParams.PodSelectorLabels,
-		Replicas:          deployParams.Replicas,
+
+	podTemplateSpec, err = patchForPolicy(podTemplateSpec, podTemplateParams.PodSecurityAdmissionPolicy)
+	if err != nil {
+		return nil, err
 	}
 
-	containerAnnotations, err := getContainerAnnotations(devfileObj, common.DevfileOptions{})
+	if needsPodOverrides(globalAttributes, components) {
+		patchedPodTemplateSpec, err := applyPodOverrides(globalAttributes, components, podTemplateSpec)
+		if err != nil {
+			return nil, err
+		}
+		patchedPodTemplateSpec.ObjectMeta = podTemplateSpecParams.ObjectMeta
+		podTemplateSpec = patchedPodTemplateSpec
+	}
+
+	podTemplateSpec.Spec.Containers, err = applyContainerOverrides(devfileObj, podTemplateSpec.Spec.Containers)
+	if err != nil {
+		return nil, err
+	}
+	podTemplateSpec.Spec.InitContainers, err = applyContainerOverrides(devfileObj, podTemplateSpec.Spec.InitContainers)
 	if err != nil {
 		return nil, err
 	}
-	deployParams.ObjectMeta.Annotations = mergeMaps(deployParams.ObjectMeta.Annotations, containerAnnotations.Deployment)
 
-	deployment := &appsv1.Deployment{
-		TypeMeta:   deployParams.TypeMeta,
-		ObjectMeta: deployParams.ObjectMeta,
-		Spec:       *getDeploymentSpec(deploySpecParams),
+	return podTemplateSpec, nil
+}
+
+func applyContainerOverrides(devfileObj parser.DevfileObj, containers []corev1.Container) ([]corev1.Container, error) {
+	containerComponents, err := devfileObj.Data.GetComponents(common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1.ContainerComponentType,
+		},
+	})
+	if err != nil {
+		return nil, err
 	}
 
-	return deployment, nil
+	getContainerByName := func(name string) (*corev1.Container, bool) {
```

Comment:
you do not need to introduce a new function for this purpose. 
should use https://github.com/devfile/library/blob/5985a2891bffdf9e0a4dddb46a9819381ee8ff8e/pkg/devfile/parser/data/v2/components.go#L28, and set `DevfileOptions.FilterByName = <the containerName>`